### ISSUE TITLE
bug: Error Applying migration `20240604133340_backfill_manual_scores` 

### ISSUE THREAD

**Issue by theluk** [langfuse/langfuse#2292]
### Describe the bug

While starting up langfuse, it breaks because it cannot apply migrations. 

```
2024-06-10 22:46:03 Prisma schema loaded from packages/shared/prisma/schema.prisma
2024-06-10 22:46:03 Datasource "db": PostgreSQL database "langfuse", schema "public" at "postgres:5432"
2024-06-10 22:46:03 
2024-06-10 22:46:03 204 migrations found in prisma/migrations
2024-06-10 22:46:03 
2024-06-10 22:46:03 Applying migration `20240604133340_backfill_manual_scores`
2024-06-10 22:46:03 Applying database migrations failed. This is mostly caused by the database being unavailable.
2024-06-10 22:46:03 Exiting...
2024-06-10 22:46:03 Error: ERROR: current transaction is aborted, commands ignored until end of transaction block
2024-06-10 22:46:03    0: schema_core::commands::apply_migrations::Applying migration
2024-06-10 22:46:03            with migration_name="20240604133340_backfill_manual_scores"
2024-06-10 22:46:03              at schema-engine/core/src/commands/apply_migrations.rs:91
2024-06-10 22:46:03    1: schema_core::state::ApplyMigrations
2024-06-10 22:46:03              at schema-engine/core/src/state.rs:226
```

---

I can confirm that the database *is* available, because the entry in the DB for the migrations was written

![image](https://github.com/langfuse/langfuse/assets/580946/93b5ff6e-8396-491d-8a00-42a87c48f962)

### To reproduce

1. Use the latest container `ghcr.io/langfuse/langfuse:latest`
2. Configure it properly
3. start it up

### SDK and container versions

ghcr.io/langfuse/langfuse:latest

### Are you interested to contribute a fix for this bug?

No

**Comment by marcklingen** [comment]
are there logs available (logs col of the migrations table)? what is you postgres version?

**Comment by theluk** [comment]
Thanks for getting back.

So I had to reset my k8s cluster due to this error (not sure what in general we could do to rollback) so I cannot tell you anything about the logs col in the migration table. But I am asking a colleague that had the same issue.

In general I see that we are using `clkao/postgres-plv8:12-2`.

**Comment by marcklingen** [comment]
The postgres version should be fine, thanks for sharing.

If your colleague could share error logs if there are any, that'd be really helpful as I cannot reproduce this issue.

**Comment by theluk** [comment]
So I reproduced it again in my k8s cluster, and it failed again at the same migration file.

I did a fresh setup with 2.47.0, everything migrated correctly. Then I again switched back to :latest and the same Error P3009 popped up. The server does not start, so I cannot ssh into and try to apply manually to see some more details. Any other tipps how to investigate?

**Comment by marcklingen** [comment]
Do you see any logs for this migration in addition to the ones already provided? The row for this migration in the migrations table should typically include a log.

**Comment by marcklingen** [comment]
P3009
"migrate found failed migrations in the target database, new migrations will not be applied. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve
{details}"

Can you have a look at which migration is marked as failed in the db and why? happy to try to quickly resolve this here

**Comment by theluk** [comment]
The issue seems to be the gen_random_uuid. 

I found that this migration is the only one missing. I tried applying the migration manually to the db, the error was that gen_random_uuid is not existant. 

Based on the [docs](https://www.postgresql.org/docs/current/functions-uuid.html) it seems it is implemented from version 13 upwards. 

Can we use another function or method to generate a random uuid?

**Comment by theluk** [comment]
![CleanShot 2024-06-11 at 12 19 35](https://github.com/langfuse/langfuse/assets/580946/1c609415-54b0-4bf3-b93f-ef7a0b573e4e)

👀

**Comment by marcklingen** [comment]
Thanks for sharing and sorry for the mishap here. We are looking into how to best fix this.

**Comment by maxdeichmann** [comment]
Hi @theluk thanks a lot for raising this. You were right, used a function which was not compatible with Postgres 12. As the migrations did not contain a schema change, but only manipulated data in the scores table, we were able to alter migration history in this [PR](https://github.com/langfuse/langfuse/pull/2329/files).
You can deploy the new container and it will clean up your unsuccessful state and apply everything correctly.

With this change we also ensured that you can use Langfuse with Postgres >=12 and on any postgres schema.

We released this fix in [v2.54.0](https://github.com/langfuse/langfuse/releases/tag/v2.54.0).

**Comment by theluk** [comment]
Awesome guys, good job 😀
